src/components/plots/styling/heatmap/HeatmapControls.jsx

```diff
@@ -26,18 +60,78 @@ const HeatmapControls = (props) => {
             tokenSeparators={[' ']}
             notFoundContent='No gene added yet.'
           />
+          {markerHeatmap && (
+            <>
+
+              <p>Select the number of top genes per cluster to show:</p>
+
+              <Space direction='horizontal'>
+                <InputNumber
+                  value={numGenes}
+                  min={1}
+                  max={25}
+                  step={1}
+                  data-testid='num-genes'
+                  onChange={(e) => setNumGenes(e)}
+                />
+                <Button
+                  onClick={() => onUpdate({ numGenes })}
+                  disabled={genesLoaded}
+                >
+                  Run
+                </Button>
+                {' '}
+              </Space>
+              <p>Gene labels:</p>
+              <Radio.Group
+                onChange={
+                  (e) => onUpdate({ showGeneLabels: e.target.value })
+                }
+                value={config.showGeneLabels}
+              >
+                <Radio value>Show</Radio>
+                <Radio value={false}>Hide</Radio>
+              </Radio.Group>
+            </>
+          )}
           <Button
             type='primary'
-            onClick={() => onGeneEnter([])}
+            onClick={onReset}
           >
             Reset
           </Button>
         </Space>
       </Panel>
-      <Panel header='Metadata tracks' key='11'>
+      {markerHeatmap && (
+        <>
+          <Panel header='Select Data' key='selectData'>
+            <p>Select the cell sets to show markers for:</p>
+            <Select
+              value={{
+                key: firstLetterUppercase(config.selectedCellSet),
```

Comment:
why aren't you using the key of the selectedCellSet? That should be unique, whereas we can have two or more cell sets with the same name